code/modules/mob/living/carbon/human/life.dm

```diff
@@ -331,13 +334,187 @@
 
 		oxygen_alert = max(oxygen_alert, 1)
 		return 0
-	var/obj/item/organ/internal/lungs/L = internal_organs_by_name[BP_LUNGS]
-	if(L && L.handle_breath(breath))
+
+	if(get_organ_efficiency(OP_LUNGS) && handle_breath_lungs(breath))
 		failed_last_breath = 0
 	else
 		failed_last_breath = 1
 	return 1
 
+/mob/living/carbon/human/proc/handle_breath_lungs(datum/gas_mixture/breath)
+	//vars - feel free to modulate if you want more effects that are not gained with efficiency 
+	var/breath_type = species.breath_type ? species.breath_type : "oxygen"
+	var/poison_type = species.poison_type ? species.poison_type : "plasma"
+	var/exhale_type = species.exhale_type ? species.exhale_type : 0
+
+	var/min_breath_pressure = species.breath_pressure
+
+	var/safe_exhaled_max = 10
+	var/safe_toxins_max = 0.2
+	var/SA_para_min = 1
+	var/SA_sleep_min = 5
+
+	var/lung_efficiency = get_organ_efficiency(OP_LUNGS)
+
+	if(!breath)
+		return 0
+
+	var/safe_pressure_min = min_breath_pressure // Minimum safe partial pressure of breathable gas in kPa
+	// Lung damage increases the minimum safe pressure.
```

Comment:
This means that if your lungs are in good condition, you are more likely to suffocate?

Reply:
This is a reference to the fact that you shouldn't hold your breath in low-pressure environments, as the damaged lungs hold less gas.


(https://www.businessinsider.com/how-long-human-survive-outer-space-without-spacesuit-2017-5#:~:text=Whatever%20you%20do%2C%20don't,your%20lungs%2C%20they%20will%20rupture.&text=Without%20air%20in%20your%20lungs,sending%20oxygen%20to%20your%20brain.)

Reply:
It seems that I misunderstood the meaning of that phrase when I read it yesterday, sorry.